File: SPRINGBOOTJPA/springbootjparest - initial/src/main/java/com/scarc/springbootjparest/models/Participation.java
package com.scarc.springbootjparest.models;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Builder
@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIdentityInfo(
    generator = ObjectIdGenerators.PropertyGenerator.class,
    property = "id")
public class Participation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private double price;
    private double paidAlready;
    private String comments;

    @ManyToOne
    private Event eventAt;
    @ManyToOne
    private Responsible responsible;
    @ManyToOne
    private Department wantsToParticipate;
    @ManyToOne
    private Company company;

    @OneToMany(mappedBy = "participation")
    private List<TimeSlot> timeSlots;
}

File: KOTLIN/Wirtschaftstag/Kotlin_Wirtschaftstag/androidApp/src/main/java/me/scarc/androidApp/data/Participation.kt
package me.scarc.androidApp.data

import me.scarc.androidApp.Company

class Participation(
    var id: Int,
    var price: Double,
    var paidAlready: Double,
    var comments: String,
    var eventAt: Event,
    var responsible: Responsible,
    var wantsToParticipate: Department,
    var company: Company,
    var timeSlots: List<TimeSlot>
) {

}
